semantics/linking/c-resolution.k

```diff
@@ -33,43 +39,47 @@ module LINKING-C-RESOLUTION
           <genv>... X |-> (Base:LinkBase => Base') ...</genv>
           <linkings>... .Map => Base |-> Base' ...</linkings>
           [structural]
-     rule <k> resolveCReference(X:CId) ...</k>
+     rule <k> resolveCReference(X:CId, _) ...</k>
           <external-decls>...
                X |-> (SetItem(Tu:String) => .Set) _::Set
           ...</external-decls>
           <external-defs>... X |-> _ ...</external-defs>
           <tu-id> Tu </tu-id>
           <genv>... X |-> Base:DirectBase ...</genv>
           [structural]
-     rule <k> resolveCReference(X:CId) => .K  ...</k>
+     rule <k> resolveCReference(X:CId, _) => .K  ...</k>
           <external-decls>...
                X:CId |-> .Set => .Map
           ...</external-decls>
           <external-defs>... X |-> _ ...</external-defs>
           <external-types>... X |-> _ ...</external-types>
           [structural]
-     rule <k> resolveCReference(X:CId) => .K ...</k>
+     rule <k> resolveCReference(X:CId, _) => .K ...</k>
           <external-defs> Defs:Map </external-defs>
           <external-types> Types:Map </external-types>
           requires (notBool (X in_keys(Defs)) orBool notBool (X in_keys(Types)))
                andBool isMangledName(X)
           [structural]
-     rule <k> (.K => setTranslationLoc(L) ~> EXT-UNDEF("TDR2",
-                    "No definition for symbol with external linkage: "
-                     +String showCId(X) +String "."))
-              ~> resolveCReference(X:CId)
-          ...</k>
+     rule <k>  (resolveCReference(X:CId, ErrorHandling:LinkErrorHandling) => setTranslationLoc(L)
```

Comment:
Using a single colon generates a run-time check for the sort, which I think is not necessary here. Try using `::`.

Reply:
Fixed.